stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/MediaGalleryPreviewActivity.kt

```diff
@@ -0,0 +1,1413 @@
+                                        )
+
+                                        if (abs(newTranslationX) < calculateMaxOffsetPerAxis(
+                                                imageSize.width,
+                                                currentScale,
+                                                parentSize.width
+                                            ) || zoom != DefaultZoomScale
+                                        ) {
+                                            event.changes.forEach { it.consume() }
+                                        }
+                                    } while (event.changes.any { it.pressed })
+
+                                    if (currentScale < DefaultZoomScale) {
+                                        currentScale = DefaultZoomScale
+                                    }
+                                }
+                            }
+                        }
+                        .pointerInput(Unit) {
+                            forEachGesture {
+                                awaitPointerEventScope {
+                                    awaitFirstDown()
+                                    withTimeoutOrNull(DoubleTapTimeoutMs) {
+                                        awaitFirstDown()
+                                        currentScale = when {
+                                            currentScale == MaxZoomScale -> DefaultZoomScale
+                                            currentScale >= MidZoomScale -> MaxZoomScale
+                                            else -> MidZoomScale
+                                        }
+
+                                        if (currentScale == DefaultZoomScale) {
+                                            translation = Offset(0f, 0f)
+                                        }
+                                    }
+                                }
+                            }
+                        },
+                    painter = painter,
+                    contentDescription = null
+                )
+
+                Log.d("isCurrentPage", "${page != pagerState.currentPage}")
+
+                if (pagerState.currentPage != page) {
+                    currentScale = DefaultZoomScale
+                    translation = Offset(0f, 0f)
+                }
+            }
+        }
+    }
+
+    /**
+     * Represents an individual page containing video player with media controls.
+     *
+     * @param attachment The video attachment to be played.
+     * @param pagerState The state of the pager that contains this page
+     * @param page The page an instance of this content is located on.
+     * @param onPlaybackError Handler for playback errors.
+     */
+    @Composable
+    private fun VideoPreviewContent(
+        attachment: Attachment,
+        pagerState: PagerState,
+        page: Int,
+        onPlaybackError: () -> Unit,
+    ) {
+        val context = LocalContext.current
```

Comment:
This function is ugly and has a lot of booleans.

This is a consequence of two things

1. We use a `View` which isn't necessarily Compose friendly
2. `VideoView` is severely limited in many ways.